api/envoy/config/filter/http/cache/v2alpha/cache.proto

```diff
@@ -0,0 +1,82 @@
+syntax = "proto3";
+
+package envoy.config.filter.http.cache.v2alpha;
+
+import "envoy/api/v2/route/route_components.proto";
+import "envoy/type/matcher/string.proto";
+
+import "google/protobuf/any.proto";
+
+import "udpa/annotations/migrate.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.config.filter.http.cache.v2alpha";
+option java_outer_classname = "CacheProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_migrate).move_to_package =
+    "envoy.extensions.filters.http.cache.v3alpha";
+
+// [#protodoc-title: HTTP Cache Filter]
+// [#extension: envoy.filters.http.cache]
+
+// [#next-free-field: 6]
+message CacheConfig {
+  // [#not-implemented-hide:]
+  // Modifies cache key creation by restricting which parts of the URL are included.
+  message KeyCreatorParams {
+    // If true, exclude the URL scheme from the cache key. Set to true if your origins always
+    // produce the same response for http and https requests.
+    bool exclude_scheme = 1;
+
+    // If true, exclude the host from the cache key. Set to true if your origins' responses don't
+    // ever depend on host.
+    bool exclude_host = 2;
+
+    // If *query_parameters_included* is nonempty, only query parameters matched
+    // by one or more of its matchers are included in the cache key. Any other
+    // query params will not affect cache lookup.
+    repeated api.v2.route.QueryParameterMatcher query_parameters_included = 3;
+
+    // If *query_parameters_excluded* is nonempty, query parameters matched by one
+    // or more of its matchers are excluded from the cache key (even if also
+    // matched by *query_parameters_included*), and will not affect cache lookup.
+    repeated api.v2.route.QueryParameterMatcher query_parameters_excluded = 4;
+  }
+
+  // Config specific to the cache storage implementation.
+  google.protobuf.Any typed_config = 1;
+
+  // Name of cache implementation to use, as specified in the intended HttpCacheFactory
+  // implementation. Cache names should use reverse DNS format, though this is not enforced.
+  string name = 2;
```

Comment:
I think we agreed to swap this to typed_config and not have a name at all here?

Reply:
Done.